### ISSUE TITLE
Compilation error and Segmentation fault

### ISSUE THREAD

**Issue by tindzk** [Affectiva/cpp-sdk-samples#15]
I have been trying to get the samples to work on a recent Arch Linux machine as well as Ubuntu 15.10. Unfortunately, for both the compilation fails with:

```bash                                                                      
[ 50%] Built target opencv-webcam-demo
[ 75%] Linking CXX executable video-demo
CMakeFiles/video-demo.dir/video-demo.cpp.o: In function `main':
video-demo.cpp:(.text+0x1198): undefined reference to `affdex::VideoDetector::process(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [video-demo/CMakeFiles/video-demo.dir/build.make:148: video-demo/video-demo] Error 1
make[1]: *** [CMakeFiles/Makefile2:141: video-demo/CMakeFiles/video-demo.dir/all] Error 2
make: *** [Makefile:128: all] Error 2
```

When I open `opencv-webcam-demo`, it crashes with a segmentation fault.

I have installed the following dependencies:
* OpenCV v3.1.0
* Boost v1.62.0

```bash
$ ldd opencv-webcam-demo/opencv-webcam-demo 
        linux-vdso.so.1 (0x00007ffff65cb000)
        libaffdex-native.so => /Users/tim/affdex-sdk/lib/libaffdex-native.so (0x00007f3c41e7e000)
        libopencv_xphoto.so.3.1 => /usr/lib/libopencv_xphoto.so.3.1 (0x00007f3c41c52000)
        libopencv_xobjdetect.so.3.1 => /usr/lib/libopencv_xobjdetect.so.3.1 (0x00007f3c41a3c000)
        libopencv_tracking.so.3.1 => /usr/lib/libopencv_tracking.so.3.1 (0x00007f3c41516000)
        libopencv_surface_matching.so.3.1 => /usr/lib/libopencv_surface_matching.so.3.1 (0x00007f3c412d0000)
        libopencv_structured_light.so.3.1 => /usr/lib/libopencv_structured_light.so.3.1 (0x00007f3c410c8000)
        libopencv_stereo.so.3.1 => /usr/lib/libopencv_stereo.so.3.1 (0x00007f3c40ea2000)
        libopencv_saliency.so.3.1 => /usr/lib/libopencv_saliency.so.3.1 (0x00007f3c40c7b000)
        libopencv_rgbd.so.3.1 => /usr/lib/libopencv_rgbd.so.3.1 (0x00007f3c409f7000)
        libopencv_reg.so.3.1 => /usr/lib/libopencv_reg.so.3.1 (0x00007f3c406bb000)
        libopencv_plot.so.3.1 => /usr/lib/libopencv_plot.so.3.1 (0x00007f3c404b1000)
        libopencv_optflow.so.3.1 => /usr/lib/libopencv_optflow.so.3.1 (0x00007f3c4028e000)
        libopencv_line_descriptor.so.3.1 => /usr/lib/libopencv_line_descriptor.so.3.1 (0x00007f3c40066000)
        libopencv_hdf.so.3.1 => /usr/lib/libopencv_hdf.so.3.1 (0x00007f3c3fe5b000)
        libopencv_fuzzy.so.3.1 => /usr/lib/libopencv_fuzzy.so.3.1 (0x00007f3c3fb32000)
        libopencv_dpm.so.3.1 => /usr/lib/libopencv_dpm.so.3.1 (0x00007f3c3f918000)
        libopencv_dnn.so.3.1 => /usr/lib/libopencv_dnn.so.3.1 (0x00007f3c3f44e000)
        libopencv_datasets.so.3.1 => /usr/lib/libopencv_datasets.so.3.1 (0x00007f3c3f1db000)
        libopencv_ccalib.so.3.1 => /usr/lib/libopencv_ccalib.so.3.1 (0x00007f3c3ef76000)
        libopencv_bioinspired.so.3.1 => /usr/lib/libopencv_bioinspired.so.3.1 (0x00007f3c3ed44000)
        libopencv_bgsegm.so.3.1 => /usr/lib/libopencv_bgsegm.so.3.1 (0x00007f3c3eb37000)
        libopencv_aruco.so.3.1 => /usr/lib/libopencv_aruco.so.3.1 (0x00007f3c3e8e9000)
        libopencv_videostab.so.3.1 => /usr/lib/libopencv_videostab.so.3.1 (0x00007f3c3e69c000)
        libopencv_superres.so.3.1 => /usr/lib/libopencv_superres.so.3.1 (0x00007f3c3e477000)
        libopencv_stitching.so.3.1 => /usr/lib/libopencv_stitching.so.3.1 (0x00007f3c3e1fe000)
        libopencv_photo.so.3.1 => /usr/lib/libopencv_photo.so.3.1 (0x00007f3c3de0e000)
        libboost_system.so.1.62.0 => /usr/lib/libboost_system.so.1.62.0 (0x00007f3c3dc0a000)
        libboost_filesystem.so.1.62.0 => /usr/lib/libboost_filesystem.so.1.62.0 (0x00007f3c3d9f1000)
        libboost_date_time.so.1.62.0 => /usr/lib/libboost_date_time.so.1.62.0 (0x00007f3c3d7e0000)
        libboost_regex.so.1.62.0 => /usr/lib/libboost_regex.so.1.62.0 (0x00007f3c3d4c7000)
        libboost_thread.so.1.62.0 => /usr/lib/libboost_thread.so.1.62.0 (0x00007f3c3d29e000)
        libboost_timer.so.1.62.0 => /usr/lib/libboost_timer.so.1.62.0 (0x00007f3c3d099000)
        libboost_chrono.so.1.62.0 => /usr/lib/libboost_chrono.so.1.62.0 (0x00007f3c3ce92000)
        libboost_serialization.so.1.62.0 => /usr/lib/libboost_serialization.so.1.62.0 (0x00007f3c3cc50000)
        libboost_log.so.1.62.0 => /usr/lib/libboost_log.so.1.62.0 (0x00007f3c3c97b000)
        libboost_log_setup.so.1.62.0 => /usr/lib/libboost_log_setup.so.1.62.0 (0x00007f3c3c6d9000)
        libboost_program_options.so.1.62.0 => /usr/lib/libboost_program_options.so.1.62.0 (0x00007f3c3c459000)
        libopencv_text.so.3.1 => /usr/lib/libopencv_text.so.3.1 (0x00007f3c3c13e000)
        libopencv_face.so.3.1 => /usr/lib/libopencv_face.so.3.1 (0x00007f3c3bf1e000)
        libopencv_ximgproc.so.3.1 => /usr/lib/libopencv_ximgproc.so.3.1 (0x00007f3c3bc5f000)
        libopencv_xfeatures2d.so.3.1 => /usr/lib/libopencv_xfeatures2d.so.3.1 (0x00007f3c3b9db000)
        libopencv_shape.so.3.1 => /usr/lib/libopencv_shape.so.3.1 (0x00007f3c3b7aa000)
        libopencv_video.so.3.1 => /usr/lib/libopencv_video.so.3.1 (0x00007f3c3b42c000)
        libopencv_objdetect.so.3.1 => /usr/lib/libopencv_objdetect.so.3.1 (0x00007f3c3b1ca000)
        libopencv_calib3d.so.3.1 => /usr/lib/libopencv_calib3d.so.3.1 (0x00007f3c3ae66000)
        libopencv_features2d.so.3.1 => /usr/lib/libopencv_features2d.so.3.1 (0x00007f3c3abab000)
        libopencv_ml.so.3.1 => /usr/lib/libopencv_ml.so.3.1 (0x00007f3c3a8f4000)
        libopencv_highgui.so.3.1 => /usr/lib/libopencv_highgui.so.3.1 (0x00007f3c3a6e4000)
        libopencv_videoio.so.3.1 => /usr/lib/libopencv_videoio.so.3.1 (0x00007f3c3a4b2000)
        libopencv_imgcodecs.so.3.1 => /usr/lib/libopencv_imgcodecs.so.3.1 (0x00007f3c3a157000)
        libopencv_imgproc.so.3.1 => /usr/lib/libopencv_imgproc.so.3.1 (0x00007f3c3871e000)
        libopencv_flann.so.3.1 => /usr/lib/libopencv_flann.so.3.1 (0x00007f3c384c8000)
        libopencv_core.so.3.1 => /usr/lib/libopencv_core.so.3.1 (0x00007f3c377b5000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f3c3742d000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007f3c37129000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f3c36f12000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f3c36b74000)
        libavcodec.so.55 => /Users/tim/affdex-sdk/lib/libavcodec.so.55 (0x00007f3c35c15000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f3c359ff000)
        libavutil.so.52 => /Users/tim/affdex-sdk/lib/libavutil.so.52 (0x00007f3c357ae000)
        libavformat.so.55 => /Users/tim/affdex-sdk/lib/libavformat.so.55 (0x00007f3c35434000)
        libswscale.so.2 => /Users/tim/affdex-sdk/lib/libswscale.so.2 (0x00007f3c351d0000)
        libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f3c34fcc000)
        libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f3c34daf000)
        librt.so.1 => /usr/lib/librt.so.1 (0x00007f3c34ba7000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f3c42fc5000)
        libhdf5.so.100 => /usr/lib/libhdf5.so.100 (0x00007f3c3464a000)
        libicudata.so.58 => /usr/lib/libicudata.so.58 (0x00007f3c32b4a000)
        libicui18n.so.58 => /usr/lib/libicui18n.so.58 (0x00007f3c326d0000)
        libicuuc.so.58 => /usr/lib/libicuuc.so.58 (0x00007f3c32324000)
        libboost_atomic.so.1.62.0 => /usr/lib/libboost_atomic.so.1.62.0 (0x00007f3c32122000)
        libtbb.so.2 => /usr/lib/libtbb.so.2 (0x00007f3c31ee5000)
        libGL.so.1 => /usr/lib/libGL.so.1 (0x00007f3c31c6d000)
        libgtk-x11-2.0.so.0 => /usr/lib/libgtk-x11-2.0.so.0 (0x00007f3c3162d000)
        libgdk-x11-2.0.so.0 => /usr/lib/libgdk-x11-2.0.so.0 (0x00007f3c31378000)
        libcairo.so.2 => /usr/lib/libcairo.so.2 (0x00007f3c3104a000)
        libgdk_pixbuf-2.0.so.0 => /usr/lib/libgdk_pixbuf-2.0.so.0 (0x00007f3c30e23000)
        libgobject-2.0.so.0 => /usr/lib/libgobject-2.0.so.0 (0x00007f3c30bd1000)
        libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0 (0x00007f3c308be000)
        libgtkglext-x11-1.0.so.0 => /usr/lib/libgtkglext-x11-1.0.so.0 (0x00007f3c306ba000)
        libgdkglext-x11-1.0.so.0 => /usr/lib/libgdkglext-x11-1.0.so.0 (0x00007f3c30456000)
        libdc1394.so.22 => /usr/lib/libdc1394.so.22 (0x00007f3c301df000)
        libxine.so.2 => /usr/lib/libxine.so.2 (0x00007f3c2ff6f000)
        libv4l1.so.0 => /usr/lib/libv4l1.so.0 (0x00007f3c2fd69000)
        libv4l2.so.0 => /usr/lib/libv4l2.so.0 (0x00007f3c2fb5b000)
        libavcodec.so.57 => /usr/lib/libavcodec.so.57 (0x00007f3c2e5ca000)
        libavformat.so.57 => /usr/lib/libavformat.so.57 (0x00007f3c2e198000)
        libavutil.so.55 => /usr/lib/libavutil.so.55 (0x00007f3c2df18000)
        libswscale.so.4 => /usr/lib/libswscale.so.4 (0x00007f3c2dc8e000)
        libjpeg.so.8 => /usr/lib/libjpeg.so.8 (0x00007f3c2da24000)
        libwebp.so.6 => /usr/lib/libwebp.so.6 (0x00007f3c2d7c3000)
        libpng16.so.16 => /usr/lib/libpng16.so.16 (0x00007f3c2d58d000)
        libtiff.so.5 => /usr/lib/libtiff.so.5 (0x00007f3c2d315000)
        libjasper.so.4 => /usr/lib/libjasper.so.4 (0x00007f3c2d0b9000)
        libIlmImf-2_2.so.22 => /usr/lib/libIlmImf-2_2.so.22 (0x00007f3c2cbe0000)
        libHalf.so.12 => /usr/lib/libHalf.so.12 (0x00007f3c2c99d000)
        libGLX.so.0 => /usr/lib/libGLX.so.0 (0x00007f3c2c76c000)
        libX11.so.6 => /usr/lib/libX11.so.6 (0x00007f3c2c42d000)
        libXext.so.6 => /usr/lib/libXext.so.6 (0x00007f3c2c21b000)
        libGLdispatch.so.0 => /usr/lib/libGLdispatch.so.0 (0x00007f3c2bf45000)
        libgmodule-2.0.so.0 => /usr/lib/libgmodule-2.0.so.0 (0x00007f3c2bd41000)
        libpangocairo-1.0.so.0 => /usr/lib/libpangocairo-1.0.so.0 (0x00007f3c2bb34000)
        libXfixes.so.3 => /usr/lib/libXfixes.so.3 (0x00007f3c2b92e000)
        libatk-1.0.so.0 => /usr/lib/libatk-1.0.so.0 (0x00007f3c2b708000)
        libgio-2.0.so.0 => /usr/lib/libgio-2.0.so.0 (0x00007f3c2b374000)
        libpangoft2-1.0.so.0 => /usr/lib/libpangoft2-1.0.so.0 (0x00007f3c2b15f000)
        libpango-1.0.so.0 => /usr/lib/libpango-1.0.so.0 (0x00007f3c2af13000)
        libfontconfig.so.1 => /usr/lib/libfontconfig.so.1 (0x00007f3c2accf000)
        libXrender.so.1 => /usr/lib/libXrender.so.1 (0x00007f3c2aac4000)
        libXinerama.so.1 => /usr/lib/libXinerama.so.1 (0x00007f3c2a8c1000)
        libXi.so.6 => /usr/lib/libXi.so.6 (0x00007f3c2a6b0000)
        libXrandr.so.2 => /usr/lib/libXrandr.so.2 (0x00007f3c2a4a5000)
        libXcursor.so.1 => /usr/lib/libXcursor.so.1 (0x00007f3c2a29a000)
        libXcomposite.so.1 => /usr/lib/libXcomposite.so.1 (0x00007f3c2a097000)
        libXdamage.so.1 => /usr/lib/libXdamage.so.1 (0x00007f3c29e94000)
        libpixman-1.so.0 => /usr/lib/libpixman-1.so.0 (0x00007f3c29bec000)
        libfreetype.so.6 => /usr/lib/libfreetype.so.6 (0x00007f3c2992f000)
        libEGL.so.1 => /usr/lib/libEGL.so.1 (0x00007f3c2971c000)
        libxcb-shm.so.0 => /usr/lib/libxcb-shm.so.0 (0x00007f3c29518000)
        libxcb.so.1 => /usr/lib/libxcb.so.1 (0x00007f3c292ef000)
        libxcb-render.so.0 => /usr/lib/libxcb-render.so.0 (0x00007f3c290e1000)
        libffi.so.6 => /usr/lib/libffi.so.6 (0x00007f3c28ed8000)
        libpcre.so.1 => /usr/lib/libpcre.so.1 (0x00007f3c28c65000)
        libGLU.so.1 => /usr/lib/libGLU.so.1 (0x00007f3c289e5000)
        libXmu.so.6 => /usr/lib/libXmu.so.6 (0x00007f3c287ca000)
        libXt.so.6 => /usr/lib/libXt.so.6 (0x00007f3c28561000)
        libpangox-1.0.so.0 => /usr/lib/libpangox-1.0.so.0 (0x00007f3c28341000)
        libraw1394.so.11 => /usr/lib/libraw1394.so.11 (0x00007f3c28131000)
        libusb-1.0.so.0 => /usr/lib/libusb-1.0.so.0 (0x00007f3c27f18000)
        libresolv.so.2 => /usr/lib/libresolv.so.2 (0x00007f3c27d01000)
        libnsl.so.1 => /usr/lib/libnsl.so.1 (0x00007f3c27ae9000)
        libv4lconvert.so.0 => /usr/lib/libv4lconvert.so.0 (0x00007f3c27870000)
        libswresample.so.2 => /usr/lib/libswresample.so.2 (0x00007f3c27653000)
        libva.so.1 => /usr/lib/libva.so.1 (0x00007f3c27433000)
        libxvidcore.so.4 => /usr/lib/libxvidcore.so.4 (0x00007f3c27121000)
        libx265.so.95 => /usr/lib/libx265.so.95 (0x00007f3c260d2000)
        libx264.so.148 => /usr/lib/libx264.so.148 (0x00007f3c25d5c000)
        libwebpmux.so.2 => /usr/lib/libwebpmux.so.2 (0x00007f3c25b52000)
        libvpx.so.4 => /usr/lib/libvpx.so.4 (0x00007f3c2560b000)
        libvorbisenc.so.2 => /usr/lib/libvorbisenc.so.2 (0x00007f3c25358000)
        libvorbis.so.0 => /usr/lib/libvorbis.so.0 (0x00007f3c2512b000)
        libtheoraenc.so.1 => /usr/lib/libtheoraenc.so.1 (0x00007f3c24eea000)
        libtheoradec.so.1 => /usr/lib/libtheoradec.so.1 (0x00007f3c24cd1000)
        libspeex.so.1 => /usr/lib/libspeex.so.1 (0x00007f3c24ab8000)
        libschroedinger-1.0.so.0 => /usr/lib/libschroedinger-1.0.so.0 (0x00007f3c247dc000)
        libopus.so.0 => /usr/lib/libopus.so.0 (0x00007f3c2458b000)
        libopenjp2.so.7 => /usr/lib/libopenjp2.so.7 (0x00007f3c2434b000)
        libopencore-amrwb.so.0 => /usr/lib/libopencore-amrwb.so.0 (0x00007f3c24137000)
        libopencore-amrnb.so.0 => /usr/lib/libopencore-amrnb.so.0 (0x00007f3c23f0d000)
        libmp3lame.so.0 => /usr/lib/libmp3lame.so.0 (0x00007f3c23c96000)
        libgsm.so.1 => /usr/lib/libgsm.so.1 (0x00007f3c23a8b000)
        liblzma.so.5 => /usr/lib/liblzma.so.5 (0x00007f3c23865000)
        libssh.so.4 => /usr/lib/libssh.so.4 (0x00007f3c235e9000)
        libmodplug.so.1 => /usr/lib/libmodplug.so.1 (0x00007f3c2325c000)
        libbluray.so.1 => /usr/lib/libbluray.so.1 (0x00007f3c23019000)
        libgnutls.so.30 => /usr/lib/libgnutls.so.30 (0x00007f3c22ce0000)
        libgmp.so.10 => /usr/lib/libgmp.so.10 (0x00007f3c22a4d000)
        libbz2.so.1.0 => /usr/lib/libbz2.so.1.0 (0x00007f3c2283d000)
        libvdpau.so.1 => /usr/lib/libvdpau.so.1 (0x00007f3c22639000)
        libva-drm.so.1 => /usr/lib/libva-drm.so.1 (0x00007f3c22436000)
        libva-x11.so.1 => /usr/lib/libva-x11.so.1 (0x00007f3c22230000)
        libImath-2_2.so.12 => /usr/lib/libImath-2_2.so.12 (0x00007f3c2201e000)
        libIex-2_2.so.12 => /usr/lib/libIex-2_2.so.12 (0x00007f3c21dfe000)
        libIlmThread-2_2.so.12 => /usr/lib/libIlmThread-2_2.so.12 (0x00007f3c21bf7000)
        libmount.so.1 => /usr/lib/libmount.so.1 (0x00007f3c219ad000)
        libharfbuzz.so.0 => /usr/lib/libharfbuzz.so.0 (0x00007f3c2174c000)
        libthai.so.0 => /usr/lib/libthai.so.0 (0x00007f3c21542000)
        libexpat.so.1 => /usr/lib/libexpat.so.1 (0x00007f3c21318000)
        libXau.so.6 => /usr/lib/libXau.so.6 (0x00007f3c21114000)
        libXdmcp.so.6 => /usr/lib/libXdmcp.so.6 (0x00007f3c20f0e000)
        libSM.so.6 => /usr/lib/libSM.so.6 (0x00007f3c20d06000)
        libICE.so.6 => /usr/lib/libICE.so.6 (0x00007f3c20ae9000)
        libudev.so.1 => /usr/lib/libudev.so.1 (0x00007f3c4314e000)
        libsoxr.so.0 => /usr/lib/libsoxr.so.0 (0x00007f3c20872000)
        libogg.so.0 => /usr/lib/libogg.so.0 (0x00007f3c2066b000)
        liborc-0.4.so.0 => /usr/lib/liborc-0.4.so.0 (0x00007f3c203ed000)
        libcrypto.so.1.0.0 => /usr/lib/libcrypto.so.1.0.0 (0x00007f3c1ff75000)
        libxml2.so.2 => /usr/lib/libxml2.so.2 (0x00007f3c1fc0d000)
        libp11-kit.so.0 => /usr/lib/libp11-kit.so.0 (0x00007f3c1f9a8000)
        libtasn1.so.6 => /usr/lib/libtasn1.so.6 (0x00007f3c1f795000)
        libnettle.so.6 => /usr/lib/libnettle.so.6 (0x00007f3c1f55d000)
        libhogweed.so.4 => /usr/lib/libhogweed.so.4 (0x00007f3c1f328000)
        libdrm.so.2 => /usr/lib/libdrm.so.2 (0x00007f3c1f118000)
        libblkid.so.1 => /usr/lib/libblkid.so.1 (0x00007f3c1eed5000)
        libuuid.so.1 => /usr/lib/libuuid.so.1 (0x00007f3c1ecd0000)
        libgraphite2.so.3 => /usr/lib/libgraphite2.so.3 (0x00007f3c1eaa4000)
        libdatrie.so.1 => /usr/lib/libdatrie.so.1 (0x00007f3c1e89c000)
        libcap.so.2 => /usr/lib/libcap.so.2 (0x00007f3c1e698000)
        libgomp.so.1 => /usr/lib/libgomp.so.1 (0x00007f3c1e46a000)
```

**Comment by riomus** [comment]
I am also using Arch. Be aware that the documentation states that you must use OpenCV 2.4.You can install it from AUR. Unfortunately even with that version, probably because of GCC and other compiling software that has versions never then used by library, compilation fail on linking process. My current solution is using docker container for execution of software that use Affectiva.

**Comment by SNBasti** [comment]
@riomus can you provide the docker container or more information on how to solve the problem? 
I am issuing similar errors as those described above. I neither can run the examples nor use the Affectiva SDK in any of my projects. Kind of frustrating. 

I am using Ubuntu 16.04 with gcc 5.4. on 64 bit machine with openCV Version 3.1.0. 
Thanks for your help.

**Comment by riomus** [comment]
Here you can find image https://github.com/riomus/affectiva-cpp-sdk-samples-docker . I will push it to docker hub so running it will take less time.

**Comment by jbetzend** [comment]
Is there no hope of getting the code updated so that it won't fail under newer gcc versions? Ubuntu 14.04 hasn't been the LTS for quite some time now...

**Comment by umangmehta12** [comment]
@jbetzend you can use the newer gcc for the same sample but you will have to use the same dependencies.

**Comment by umangmehta12** [comment]
@riomus thanks for creating the docker image, I plan to create one for the latest gcc soon.

**Comment by nikhilvs** [comment]
i am also hitting same issue.
Here is my make and cmake output :
osboxes@osboxes:~/build$ cmake -DOpenCV_DIR=/usr/ -DBOOST_ROOT=/usr/ -DAFFDEX_DIR=$HOME/affdex-sdk $HOME/sdk-samples
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting up compiler warnings
-- Bitness detected: 64
-- 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   system
--   filesystem
--   date_time
--   regex
--   thread
--   timer
--   chrono
--   serialization
--   log
--   log_setup
--   program_options
--   atomic
-- 
-- ------- SUMMARY -------
-- Boost version found   = 1.58.0 (105800)
--   - system         /usr/lib/x86_64-linux-gnu/libboost_system.so
--   - filesystem     /usr/lib/x86_64-linux-gnu/libboost_filesystem.so
--   - date_time      /usr/lib/x86_64-linux-gnu/libboost_date_time.so
--   - regex          /usr/lib/x86_64-linux-gnu/libboost_regex.so
--   - thread         /usr/lib/x86_64-linux-gnu/libboost_thread.so
--   - timer          /usr/lib/x86_64-linux-gnu/libboost_timer.so
--   - chrono         /usr/lib/x86_64-linux-gnu/libboost_chrono.so
--   - serialization  /usr/lib/x86_64-linux-gnu/libboost_serialization.so
--   - log            /usr/lib/x86_64-linux-gnu/libboost_log.so
--   - log_setup      /usr/lib/x86_64-linux-gnu/libboost_log_setup.so
--   - program_options
--                    /usr/lib/x86_64-linux-gnu/libboost_program_options.so
-- 
-- Affdex
-- /home/osboxes/affdex-sdk/lib/libaffdex-native.so
-- 
-- OpenCV version found   = 2.4.9 (2.4.9.1)
-- OpenCV_LIB_DIR         = /usr/share/OpenCV/lib
--   - objdetect      opencv_objdetect
--   - ml             opencv_ml
--   - imgproc        opencv_imgproc
--   - highgui        opencv_highgui
--   - core           opencv_core
-- 
-- Apps identified for building:
--    - video-demo
-- Configuring done
-- Generating done
-- Build files have been written to: /home/osboxes/build


osboxes@osboxes:~/build$ make

make output :

[ 50%] Linking CXX executable video-demo
CMakeFiles/video-demo.dir/video-demo.cpp.o: In function `main':
video-demo.cpp:(.text+0x1093): undefined reference to `affdex::VideoDetector::process(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
video-demo/CMakeFiles/video-demo.dir/build.make:126: recipe for target 'video-demo/video-demo' failed
make[2]: *** [video-demo/video-demo] Error 1
CMakeFiles/Makefile2:85: recipe for target 'video-demo/CMakeFiles/video-demo.dir/all' failed
make[1]: *** [video-demo/CMakeFiles/video-demo.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by adennie** [comment]
@nikhilvs see https://stackoverflow.com/questions/37004976/undefined-reference-to-processstd-cxx11basic-string-when-compiling-a

**Comment by nikhilvs** [comment]
I saw that link, it didn’t help.
Anyway I changed my Ubuntu version from 16.04 to 14.04. Now this issue is resolved, got into another one!!

root@ip-15-0-52-159:~/video_analytics# ll
total 16
drwxr-xr-x 4 root root 4096 Jun  9 12:22 ./
drwx------ 8 root root 4096 Jun  9 12:24 ../
drwxr-xr-x 2 root root 4096 Jun  9 12:45 data/
drwxr-xr-x 2 root root 4096 Jun  9 12:24 videos/
root@ip-15-0-52-159:~/video_analytics# video-demo   -i ./videos/baby1.mp4  --data /root/video_analytics/data/test
libdc1394 error: Failed to initialize libdc1394
Max num of faces set to: 1
Face detector mode set to: SMALL_FACES
Failed to initialize, invalid data folder
root@ip-15-0-52-159:~/video_analytics#


1.       libdc1394 error: Failed to initialize libdc1394    (google say this is not harmful)

2.       Failed to initialize, invalid data folder

Even though I have created data folder, it is telling ‘Failed to initialize, invalid data folder’. What am I missing ?



--
Regards,
Nikhil VS

**Comment by adennie** [comment]
@nikhilvs what's in /root/video_analytics/data/test?

**Comment by nikhilvs** [comment]
Its empty file

**Comment by nikhilvs** [comment]
I tried giving path as Directory and File(hence test file). Both gives same error.

**Comment by adennie** [comment]
the --data param should identify the location of affdex data directory, which comes with the SDK.  It should contain subfolders like: noctali0.83, haarcascade, CT_pack, and production_config.

**Comment by nikhilvs** [comment]
Thanks.

Any chance your library supports with latest GCC (> 5)  issue I mentioned before, because we would like to use latest OS version and libraries.


--
Regards,
Nikhil VS

**Comment by adennie** [comment]
Yes, it's on our to-do list...  sorry for the inconvenience in the mean time.

**Comment by manav95** [comment]
0
down vote
favorite
I have tried to compile the affdex video-demo with the following commands:

`cmake -DCMAKE_C_COMPILER=/usr/bin/gcc-4.8 -DCMAKE_CXX_COMPILER=/usr/bin/g++-4.8 -DAFFDEX_DIR=$HOME/affdexUb $HOME/sdk-samples -DOpenCV_DIR=/usr/share/OpenCV -DBOOST_ROOT=/root/boost_1_55_0/stage/lib/`

`make video-demo`
Here are the results:

```
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/gcc-4.8
-- Check for working C compiler: /usr/bin/gcc-4.8 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/g++-4.8
-- Check for working CXX compiler: /usr/bin/g++-4.8 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting up compiler warnings
-- Bitness detected: 64
-- 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Boost version: 1.61.0
-- Found the following Boost libraries:
--   system
--   filesystem
--   date_time
--   regex
--   thread
--   timer
--   chrono
--   serialization
--   log
--   log_setup
--   program_options
--   atomic
-- 
-- ------- SUMMARY ------- 
-- Boost version found   = 1.61.0 (106100)
--   - system         /usr/lib/x86_64-linux-gnu/libboost_system.so
--   - filesystem     /usr/lib/x86_64-linux-gnu/libboost_filesystem.so
--   - date_time      /usr/lib/x86_64-linux-gnu/libboost_date_time.so
--   - regex          /usr/lib/x86_64-linux-gnu/libboost_regex.so
--   - thread         /usr/lib/x86_64-linux-gnu/libboost_thread.so
--   - timer          /usr/lib/x86_64-linux-gnu/libboost_timer.so
--   - chrono         /usr/lib/x86_64-linux-gnu/libboost_chrono.so
--   - serialization  /usr/lib/x86_64-linux-gnu/libboost_serialization.so
--   - log            /usr/lib/x86_64-linux-gnu/libboost_log.so
--   - log_setup      /usr/lib/x86_64-linux-gnu/libboost_log_setup.so
--   - program_options
--                    /usr/lib/x86_64-linux-gnu/libboost_program_options.so
-- 
-- Affdex
-- /root/affdexUb/lib/libaffdex-native.so
-- 
-- OpenCV version found   = 2.4.9 (2.4.9.1)
-- OpenCV_LIB_DIR         = /usr/share/OpenCV/lib
--   - objdetect      opencv_objdetect
--   - ml             opencv_ml
--   - imgproc        opencv_imgproc
--   - highgui        opencv_highgui
--   - core           opencv_core
-- 
-- Apps identified for building:
--    - opencv-webcam-demo
--    - video-demo
-- Configuring done
-- Generating done
-- Build files have been written to: /root/build

```

My issues are two-fold; I am not able to get cmake to find the boost 1.55 libs that I compiled from source to get past the c++ 11 issue; it instead keeps pointing to a non-existing location for Boost 1.61 that I deleted a while ago. Im also not sure if its pointing to the .so files under /usr/share/OpenCV/lib, which are there as well. The issue with the nonexistent boost libraries led to the following error, and I am not sure how to resolve this issue of not being able to Cmake to freaking point to the right location on my system. How do I get CMake to cooperate with me?

**Comment by adennie** [comment]
@manav95 did you resolve your cmake issue?  If so, please let us know; if not, please create a separate issue, as it doesn't appear to be related to this issue, as described by the original poster above.  Thanks.

**Comment by mrkaworu** [comment]
I came across the same "undefined reference" error in Ubuntu 16.04 64bits using boost 1.59, OpenCV 2.4.11, etc. 

I downloaded gcc-4.8 and used it but then the makefile was unable to find the opencv installation for some reason. There was always at least one component not properly working. So I gave up and started from scratch on a fresh Ubuntu 14.04 installation using exactly the versions described in the readme and It worked at the first try.
